client/src/app/components/menu/menu-item.ts

```diff
@@ -11,16 +11,16 @@ export class MenuItem {
     public readonly children: MenuItem[];
 
     constructor(name: string,
-                streams?: MenuItemUpdates = {},
-                children?: MenuItem[] = []) {
```

Comment:
Yes, you can't assign a default value to an optional argument.
It's either `arr?: Array` or `arr: Array = []`.
I picked the first one without any particular technical reason.